queries/schedules.py

```diff
@@ -90,271 +95,831 @@ def help_text(lemma: str) -> str:
+
+        day_diff = qdatetime.date() - datetime.date.today()
+
+        if day_diff == datetime.timedelta(days=1):
+            showtime += " á morgun"
+        elif day_diff == datetime.timedelta(days=-1):
+            showtime += " í gær"
+        elif day_diff != datetime.timedelta(days=0):
+            showtime += qdatetime.strftime(" %-d. %B")
+
+    if curr_prog:
+        curr_title, curr_desc = _extract_title_and_desc(curr_prog, station)
+
+        answer = f"Á {channel}{showtime} {showing} {curr_title}."
+
+        if curr_desc != "":
+            answer += " " + _clean_desc(curr_desc)
+
+        # Answer is valid for station until the program starts
+        _, prog_endtime = _get_program_start_end(curr_prog, station)
+    else:
+        if is_now:
+            answer = f"Ekkert er á dagskrá á {channel} í augnablikinu."
+        else:
+            answer = f"Ekkert {'verður' if is_future else 'var'} á dagskrá á {channel}{showtime}."
+
+    voice = answer
+    return {
+        "response": {"answer": answer, "voice": voice},
+        "answer": answer,
+        "voice": voice,
+        "station": station,
+        "channel": channel,
+        "expire_time": cast(datetime.datetime, prog_endtime),
+    }
+
+
+def _get_schedule_answer(result: Result) -> _AnswerDict:
+    """Generate answer to query about current TV/radio program."""
+
+    api_channel: str = result.get("api_channel")
+    channel: str = result.get("channel")
+    station: str = result.get("station")
+    is_radio: bool = result.get("channel_type") == "radio"
+
+    now: datetime.datetime = datetime.datetime.now()
+    now_date: datetime.date = now.date()
+    now_time: datetime.time = now.time()
+
+    qdate: datetime.date = result.get("qdate") or now_date
+    qtime: datetime.time = result.get("qtime") or now_time
+    # Construct datetime from date/time in query (by default use current date/time)
+    qdt: datetime.datetime = datetime.datetime.combine(qdate, qtime)
+
+    # If exact time isn't specified (e.g. "klukkan sjö" instead of "klukkan 7:00")
+    if not result.get("exact_time"):
+        if qdt.hour < 12 and result.get("PM"):
+            # If wording specifies afternoon (e.g. "í kvöld", "eftir hádegi")
+            qdt += datetime.timedelta(hours=12)
+
+        elif result.get("will_be") and qdt < now:
+            # If wording implies we want schedule in future
+            # and query datetime is in past
+            # (e.g. "verður sýnt kl sjö" when current time is more than seven)
+            qdt += datetime.timedelta(hours=12)
+
+    get_next: bool = result.get("get_next", False)
+
+    # Fetch schedule data from API or cache.
+    sched: _SchedType = _query_schedule_api(api_channel, station, qdt.date())
+
+    if len(sched) == 0:
+        date: str = ""
+        # Add date if not asking about today
+        if qdt.date() != now_date:
+            with changedlocale(category="LC_TIME"):
+                date = qdt.strftime(" %-d. %B")
```

Comment:
Use your new icestrftime function here instead of changelocale?

Reply:
The icestrftime function I created doesn't work because of some error when calling datetime.strftime with a string containing icelandic characters. It might be a pypy specific thing.